inc/ThirdParty/Plugins/CDN/Cloudflare.php

```diff
@@ -358,6 +401,42 @@ public function purge_cloudflare_partial( $urls ) {
 		$this->facade->purge_urls( $post_ids );
 	}
 
+	/**
+	 * Purges Cloudflare cache after automatic cache purge (cache lifespan)
+	 *
+	 * @since 3.20.2
+	 *
+	 * @param array $deleted Array of deleted cache data.
+	 * @param array $args    Array with lifespan and file_age_limit data.
+	 *
+	 * @return void
+	 */
+	public function purge_cloudflare_after_automatic_purge( $deleted, $args ) {
+		if ( ! $this->is_plugin_active() ) {
+			return;
+		}
+
+		// Extract URLs from deleted cache data and purge only those specific URLs.
+		$urls_to_purge = $this->extract_urls_from_deleted_cache( $deleted );
+
+		if ( empty( $urls_to_purge ) ) {
+			// If no specific URLs found, fall back to purging everything as safety measure.
+			$this->facade->purge_everything();
+			return;
+		}
+
+		// Convert URLs to post IDs for Cloudflare purging.
+		$post_ids = array_filter( array_map( 'url_to_postid', $urls_to_purge ) );
+
+		if ( ! empty( $post_ids ) ) {
+			$this->facade->purge_urls( $post_ids );
+			return;
+		}
+
+		// If no valid post IDs found, fall back to purging everything.
+		$this->facade->purge_everything();
```

Comment:
Based on the thread [here](https://group-onecom.slack.com/archives/C08EFCYRQ2X/p1763462133128759?thread_ts=1763367090.157029&cid=C08EFCYRQ2X). We don't need a fallback